src/main/java/com/enderio/base/common/block/IndustrialInsulationBlock.java

```diff
@@ -0,0 +1,68 @@
+package com.enderio.base.common.block;
+
+import com.google.common.collect.Lists;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.util.Tuple;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.*;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.Material;
+
+import java.util.Arrays;
+import java.util.Queue;
+
+public class IndustrialInsulationBlock extends SpongeBlock {
+    private final int DIRECTIONS_COUNT = Direction.values().length;
+
+    public IndustrialInsulationBlock(Properties props) {
+        super(props);
+    }
+
+    @Override
+    protected void tryAbsorbWater(Level level, BlockPos pos) {
+        absorb(level, pos);
+    }
+
+    @Override
+    public void neighborChanged(BlockState state, Level level, BlockPos pos, Block block, BlockPos fromPos, boolean isMoving) {
+        absorb(level, pos);
+        super.neighborChanged(state, level, pos, block, fromPos, isMoving);
+    }
+
+    // Overriding sponge default behavior, easier than overriding the whole BlockSponge and removing all the BlockState code
+    protected void absorb(Level level, BlockPos pos) {
+        Queue<Tuple<BlockPos, Integer>> queue = Lists.newLinkedList();
```

Comment:
don't just copy the 1.12 code, reimplement the feature and think about new things that could interact with this
(like waterloggable blocks in vanilla or water&lavaloggable blocks in mekanism or seagrass etc, try to absorb the general fluidstate instead)

Reply:
What I did was copying the code and adapting it to run on 1.19. I have no further knowlage about new things. So I used the `LiquidBlock` interface to validate the fluids. I really think it could be better, but I don't know how to improve this.